src/actions.js

```diff
@@ -3,13 +3,18 @@ import { scGetTracks } from './scFetch';
 
 export const addFun = createAction('ADD_FUN');
 export const appendNewTracks = createAction('APPEND_NEW_TRACKS');
+export const newTracksAppended = createAction('NEW_TRACKS_APPENDED');
+export const newTracksAppendFailed = createAction('NEW_TRACKS_APPEND_FAILED');
+export const setTrackNameFilter = createAction('SET_TRACK_NAME_FILTER');
+export const setArtistNameFilter = createAction('SET_ARTIST_NAME_FILTER');
 
 // async
 export const getNewTracks = () => {
   return dispatch => {
+    dispatch(appendNewTracks());
     scGetTracks().then(tracks =>
-      dispatch(appendNewTracks(tracks)).catch(error =>
-        dispatch(appendNewTracks(error))
+      dispatch(newTracksAppended(tracks)).catch(error =>
+        dispatch(newTracksAppendFailed(error))
```

Comment:
Non-blocking: You *can* use the same action creator in both then/catch. If you pass an Error subclass as the payload (such as one `catch` gives you), it'll set `action.error` to `true`. You could then have one reducer per action, with just a different bit of unpacking:

```js
{
  [newTracksAppend]: (state, {payload, error}) => { /* if error else blah */ }
}
```

Either separate actions or a single action with `error` are acceptable approaches. FSA/RSA  [lean toward using one](https://github.com/redux-utilities/flux-standard-action).